### PR TITLE
renamed incomeDetailsKey => incomeDetailsType + added nonMonetaryContributions

### PR DESCRIPTION
**Pull request by taoeffect**
- renamed incomeDetailsKey => incomeDetailsType
- added nonMonetaryContributions to group profile

### CODE REVIEW

**Review by sandrina-p — APPROVED** [review]

**Comment by taoeffect** [comment]
**NOTE: this is an exceptional merge that violates Rule 7 of our contributions guidelines because it's during a hackathon and I have approval from @sandrina-p to merge it, and @sandrina-p cannot merge it herself because the build is failing, but it's an error unrelated to this PR.**

**Pull request merged**
